typescript/infra/scripts/helloworld/kathy.ts

```diff
@@ -7,40 +8,37 @@ import {
   InterchainGasCalculator,
 } from '@abacus-network/sdk';
 
-import { submitMetrics } from '../../src/utils/metrics';
-import { sleep } from '../../src/utils/utils';
+import { debug, error, log } from '../../src/utils/logging';
+import { startMetricsServer } from '../../src/utils/metrics';
+import { diagonalize, sleep } from '../../src/utils/utils';
 import { getCoreEnvironmentConfig, getEnvironment } from '../utils';
 
 import { getApp } from './utils';
 
-const constMetricLabels = {
-  // this needs to get set in main because of async reasons
-  abacus_deployment: '',
-  abacus_context: 'abacus',
-};
-
 const metricsRegister = new Registry();
-const messagesSendStatus = new Gauge({
+const messagesSendCount = new Counter({
   name: 'abacus_kathy_messages',
   help: 'Whether messages which have been sent from one chain to another successfully; will report 0 for unsuccessful and 1 for successful.',
   registers: [metricsRegister],
-  labelNames: [
-    'origin',
-    'remote',
-    ...(Object.keys(constMetricLabels) as (keyof typeof constMetricLabels)[]),
-  ],
+  labelNames: ['origin', 'remote', 'status'],
+});
+const currentPairingIndexGauge = new Gauge({
+  name: 'abacus_kathy_pairing_index',
+  help: 'The current message pairing index kathy is on, this is useful for seeing if kathy is always crashing around the same pairing as pairings are deterministically ordered.',
```

Comment:
what is "pairing index"? can we just surface source and destination chain name in the metric?

Reply:
No, we can't. The purpose of the metric is more than anything just to be able to see if it is going through the indexes at an expected rate.